File: server/src/drivers/typescript/driver.ts
const { stdin, stdout, stderr, exit } = require('process')

type datatype =
  | 'Boolean'
  | 'String'
  | 'Integer'
  | 'Integer[]'
  | 'String[]'
  | 'Integer[][]'
  | 'String[][]'

export interface Param {
  name: string
  type: datatype
}

let inputsData = ''
let funcName = ''
let params: Param[] = []
let args = ''

stdin.on('data', (data: Buffer) => {
  inputsData += data.toString('utf-8')
})

stdin.on('end', () => {
  let paramStr: string = '';
  [funcName, paramStr, args] = inputsData.trim().split('|')
  params = JSON.parse(paramStr.replace(/\\/g, ''))
  main()
})


function convertArg(type: datatype, input: string) {
  let temp;

  try {
    switch (type) {
      case 'String[][]':
      case 'Integer[][]':
        temp = JSON.parse(input)
        if(!temp.every((el: any) => el instanceof Array)) throw TypeError();
        return temp

      case 'Integer[]':
      case 'String[]':
        return JSON.parse(input)

      case 'Integer':
        temp = parseInt(input)
        if(Number.isNaN(temp)) throw TypeError();
        return temp
        
      case 'Boolean':
        return input === 'true' ? true : false
      default:
        return input
    }
  } catch (err) {
    stderr.write(`${input} is not a valid value of type ${type}`)
    exit(1)
  }
}


function main() {
  const inputs = args.split(/\n/)
  const convertedArgs = []

  for (let i = 0; i < inputs.length; i++) {
    const param = params[i]
    const input = inputs[i]

    convertedArgs.push(convertArg(param.type, input))
  }

  const output = eval(funcName)(...convertedArgs)

  stdout.write(output.toString())
}

File: server/src/drivers/python/driver.py
from sys import stderr, stdin, stdout, exit
from typing import List
from json import loads, dumps 
import re

class Solution: pass

input = stdin.read()

funcName = ''
params = {}
args = []

inputs = input.split("|")

funcName = inputs[0]
params = loads(inputs[1].replace("\\", ""))
args = inputs[2].split("\n")

def convertToType(type: str, arg: str):
    try:
        if type in ['Integer[]', 'String[]']:
            return loads(arg)
        elif type in ['Integer[][]', 'String[][]']:
            temp = loads(arg)

            if (len(filter(lambda el: isinstance(el, list), temp)) != len(temp)):
                raise TypeError()
        
            return temp

        elif type == 'Integer':
            return int(arg)
        elif type == 'Boolean':
            return  True if arg == 'true' else False
        else: 
            return arg
    except:
        stderr.write('% s is not a valid value of type % s' % (arg, type))
        exit(1)

def main():
    convertedArgs = []

    for i in range(len(params)):
        param = params[i]
        arg = args[i]

        convertedArgs.append(convertToType(param['type'], arg))

    solution = Solution()

    output = getattr(solution, funcName)(*convertedArgs)

    if(isinstance(output, list)):
        stdout.write(re.sub("\s|\"", "", dumps(output)))
    else:
        stdout.write(str(output))

main()
